nullaway/src/main/java/com/uber/nullaway/generics/CheckIdenticalNullabilityVisitor.java

```diff
@@ -75,7 +81,7 @@ public Boolean visitClassType(Type.ClassType lhsType, Type rhsType) {
     // If there is an enclosing type (for non-static inner classes), its type argument nullability
     // should also match.  When there is no enclosing type, getEnclosingType() returns a NoType
     // object, which gets handled by the fallback visitType() method
-    return lhsType.getEnclosingType().accept(this, rhsType.getEnclosingType());
+    return lhsType.getEnclosingType().accept(this, rhsTypeAsSuper.getEnclosingType());
```

Comment:
Was this working before? What's the change in semantics between using `rhsType` vs `rhsTypeAsSuper` here to get the enclosing type? (Just curious)

Reply:
This is a very good question and I went down a bit of a rabbit hole trying to figure it out :-)  The change was suggested by Code Rabbit and seemed harmless, but as I dug more I couldn't construct a test where it actually mattered.  If we just use `rhsType`, then when comparing enclosing types, we should recursively call `asSuper()` and everything _should_ work out.  I undid the change but added a comment to this effect.

While I was digging into this I found https://github.com/uber/NullAway/issues/1274 and https://github.com/typetools/checker-framework/issues/7239 🙂